### PR TITLE
feat: improve deep equality check failing message for toBe

### PR DESCRIPTION
**Pull request by TommyDew42**
## What does this PR do?

A PR for issue #1290.

We would like to have a message to remind users they should use deep equality checks, when object comparison fails with `toBe`. The error message would look like this now:
Screenshot 2022-05-27 at 12 12 49 PM

The implementation is similar to how [jest](https://github.com/facebook/jest/blob/main/packages/expect/src/matchers.ts#L85-L124) does it.

## How the change is tested?
Unit tests

### CODE REVIEW

**Comment by ghiscoding** [comment]
Thanks for working on the issue that I opened, however I still prefer how Jest shows the error, the text is more aligned especially around the expectations of expected/received. Also I see you added square brackets around the "serializes to the same string" but I find that a little bit confusing, I didn't know (until I looked at your code change) that the bracket was something that you added, I wasn't sure if the text inside it was dynamic or not (it's not) and that made it confusing, it's also not an array so why use brackets and not double quotes? Nonetheless, it's a great improvement compare to current code, so I'd be curious to see what the core team will reply. Thanks again for working on the issue

I like how Jest shows it, everything is aligned and more readable
```sh
expect(received).toBe(expected) // Object.is equality

    If it should pass with deep equality, replace "toBe" with "toStrictEqual"

    Expected: [{"age": 30, "id": 123, "name": "John"}]
    Received: serializes to the same string

      at Object.<anonymous> (/src/index.spec.ts:34:50)
```

File: packages/vitest/src/integrations/chai/jest-expect.ts [hunk 1]
```diff
@@ -129,9 +129,44 @@ export const JestChaiExpect: ChaiPlugin = (chai, utils) => {
...
+        const toEqualPass = jestEquals(
+          actual,
+          expected,
+          [iterableEquality],
+        )
+
+        if (toEqualPass) {
+          reminderMessage = 'If it should pass with deep equality, replace "toBe" with "toEqual"\n\n'
+
+          toBeMessage = `${reminderMessage}expected #{this} to be [serializes to the same string] // Object.is equality`
```
Comment (sheremet-va) [inline packages/vitest/src/integrations/chai/jest-expect.ts:162]: `expected to be` should be at the top. this is error message.

`expected/received` block is rendered independently

**Review by sheremet-va — CHANGES_REQUESTED** [review]

**Comment by TommyDew42** [comment]
@ghiscoding @sheremet-va 

Does the current error message look good to us?
Screenshot 2022-06-02 at 1 03 00 AM

**Comment by ghiscoding** [comment]
> @ghiscoding @sheremet-va
> 
> Does the current error message look good to us? Screenshot 2022-06-02 at 1 03 00 AM

Yeah this is much closer to Jest message, love it

File: packages/vitest/src/integrations/chai/jest-expect.ts (hunk 1 above)
Comment (TommyDew42) [reply packages/vitest/src/integrations/chai/jest-expect.ts]: Solved in commit [9cc61](https://github.com/vitest-dev/vitest/pull/1383/commits/9cc61ea2bdba3ed7cab1a93bb9020ed98e8356c9).

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**